env_spec.py

```diff
@@ -19,138 +28,216 @@
+        if env_spec_entry["comment"] is not None:
+            ret_str += f"<small>{env_spec_entry['comment']}</small>\n"
+
+        html_output += ret_str
+    return html_output
+
+
+def parse(env_spec_text):
     """
-    Creates html string from env_variables (field, type). If env_var_type is a restricted choice,
-    calls create_list to create the list of strs from env_var_type string. Returns the string.
+    Parse function creates a list of dictionaries by splitting env_spec_text by
+    \n. Every line will be a dictionary.The keys of dictionary are:
+    name, type, choices, default_value, comment.
+    Returns the env_spec listm or empty list [], for exception.
     """
+    env_spec_list = []
 
-    env_var_field_lower = env_var_field.lower()
-    env_var_type = env_var_type or "text"
+    alphanumeric_that_does_not_start_with_digit = r"^[A-Z_][0-9A-Z_]*$"
+    name_regex = r"^(.+)\:(.+)$"
+    choices_regex = r"^(.+)\]"
+    default_values_regex = r"^(.+)\=(.+)$"
+    line_comment_regex = r"^#"
+    comment_regex = r"^(.+)\#(.+)$"
 
-    if "[" not in env_var_type:
-        if "=" in env_var_type:
-            env_var_type, value = env_var_type.split("=")
-            value = value.strip()
+    lines = env_spec_text.split("\n")
 
-            ret_str = (
-                f'<label for="env_spec_{env_var_field_lower}">{env_var_field}</label>\n'
-                f'<input id="env_spec_{env_var_field_lower}" name="{env_var_field_lower}" type="{env_var_type}" value="{value}" />\n'
-            )
+    for line in lines:
+        line_dict = {}
+
+        line_comment_match = re.match(line_comment_regex, line)
+
+        if line_comment_match:
+            continue
+        comment_match = re.match(comment_regex, line)
+
+        if comment_match:
+            line = comment_match.groups()[0]
+            comment = comment_match.groups()[1]
+            line_dict["comment"] = comment
         else:
-            ret_str = (
-                f'<label for="env_spec_{env_var_field_lower}">{env_var_field}</label>\n'
-                f'<input id="env_spec_{env_var_field_lower}" name="{env_var_field_lower}" type="{env_var_type}" />\n'
-            )
-    else:
-        ret_str = ""
-        try:
-            restr_choices, value = env_var_type.split("=")
-            value = value.strip()
-            env_var_type_list = create_list(restr_choices)
-        except:
-            env_var_type_list = create_list(env_var_type)
-
-        ret_str += (
-            f'<label for="env_spec_{env_var_field_lower}">{env_var_field}</label>\n'
-            f'<select id="env_spec_{env_var_field_lower}" name="{env_var_field_lower}">\n'
-        )
-
-        for line in env_var_type_list:
-            if "=" in env_var_type and value == line:
-                ret_str += f'\t<option value="{line}"selected>{line}</option>\n'
-            else:
-                ret_str += f'\t<option value="{line}">{line}</option>\n'
+            line_dict["comment"] = None
```

Comment:
I propose removing this whole `else` clause and just initialize the comment as `None` when [creating the object above](
https://github.com/sourcelair/env-spec-py/pull/16/files#diff-662ae986ce9e28e035956190e3f42976R128).